### PR TITLE
docs: Update FAQ about known types (#6008)

### LINKED ISSUE #6008: Identical time values shouldn't be treated as different or the FAQ should suggest using `knownTypeFields`
**Workaround**
See https://github.com/argoproj/argo-cd/issues/6008#issuecomment-1346433001

**Not a general suport forum**
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

**Checklist**

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

The prometheus-community helm chart is using a short time format, and cert-manager is using golang's Duration.string() function which writes out long time format (including hours and minutes).

duration: 8760h0m0s and duration: 8760h are treated as different.

This triggers annoying sync behaviors

https://github.com/prometheus-community/helm-charts/pull/750

**To Reproduce**

install cert-manager using:
```yaml
name: cert-manager
apiVersion: v2
appVersion: "1.0"
description: Cert Manager
version: 1.0.4
dependencies:
  - name: cert-manager
    version: 1.0.4
    repository: https://charts.jetstack.io
```

install prometheus operator using:
```yaml
name: prometheus-operator
apiVersion: v2
appVersion: "1.0"
description: Prometheus operator
version: 14.4.0
dependencies:
  - name: kube-prometheus-stack
    version: 14.4.0
    repository: https://prometheus-community.github.io/helm-charts
```

Possibly add some values to ensure the certificate is created:

values.yaml:
```yaml
kube-prometheus-stack:
  prometheusOperator:
    tlsProxy:
      enabled: false
    admissionWebhooks:
      certManager:
        enabled: true
      enabled: false
```

https://github.com/prometheus-community/helm-charts/pull/750

**Expected behavior**

ideally `duration` fields would be understood as times and compared as times.

**Screenshots**

image

\# | Live | # | Desired
-- | -- | -- | --
56 |  prometheus-kube-prometheus-operator.metrics | 56 |  prometheus-kube-prometheus-operator.metrics
57 |  prometheus-kube-prometheus-operator.metrics.svc | 57 |  prometheus-kube-prometheus-operator.metrics.svc
58 | duration: 8760h0m0s | 58 | duration: 8760h
59 | issuerRef: | 59 | issuerRef:
60 | name: prometheus-kube-prometheus-root-issuer | 60 | name: prometheus-kube-prometheus-root-issuer

**Version**

```shell
{
    "Version": "v2.0.0+d085636",
    "BuildDate": "2021-03-30T19:10:08Z",
    "GitCommit": "d085636fd72c170d0553a2db1f0de8e6ea0df398",
    "GitTreeState": "clean",
    "GoVersion": "go1.16",
    "Compiler": "gc",
    "Platform": "linux/amd64",
    "KsonnetVersion": "v0.13.1",
    "KustomizeVersion": "v3.9.4 2021-02-09T19:22:10Z",
    "HelmVersion": "v3.5.1+g32c2223",
    "KubectlVersion": "v0.20.4",
    "JsonnetVersion": "v0.17.0"
}
```

**Logs**

```
Paste any relevant application logs here.
```

### PR DESCRIPTION
**Pull request by andrii-korotkov-verkada**
Closes #6008

Make more emphasis on the issue users may be searching for, update wording slightly and another example.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [x] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -151,20 +151,20 @@ The default polling interval is 3 minutes (180 seconds) with a configurable jitt
 You can change the setting by updating the `timeout.reconciliation` value and the `timeout.reconciliation.jitter` in the [argocd-cm](https://github.com/argoproj/argo-cd/blob/2d6ce088acd4fb29271ffb6f6023dbb27594d59b/docs/operator-manual/argocd-cm.yaml#L279-L282) config map. If there are any Git changes, Argo CD will only update applications with the [auto-sync setting](user-guide/auto_sync.md) enabled. If you set it to `0` then Argo CD will stop polling Git repositories automatically and you can only use alternative methods such as [webhooks](operator-manual/webhook.md) and/or manual syncs for deploying applications.
 
 
-## Why Are My Resource Limits `Out Of Sync`?
+## Why is my ArgoCD application `Out Of Sync` and keep syncing when there are no actual changes in resource limits or other fields with unit values?
 
 Kubernetes has normalized your resource limits when they are applied, and then Argo CD has then compared the version in
```
Comment (nitishfy) [inline docs/faq.md:156]:
```suggestion
Kubernetes has normalized your resource limits when they are applied, and then Argo CD compared the version in
```

File: docs/faq.md [hunk 2]
```diff
@@ -151,20 +151,20 @@ The default polling interval is 3 minutes (180 seconds) with a configurable jitt
...
-your generated manifests to the normalized one is Kubernetes - they won't match.
+your generated manifests to the normalized one is Kubernetes - they may not match.
```
Comment (nitishfy) [inline docs/faq.md:157]:
```suggestion
your generated manifests from git to the normalized one in Kubernetes cluster - they may not match.
```

**Review by nitishfy — CHANGES_REQUESTED** [review]
Overall LGTM, please address the following comments.

**Comment by andrii-korotkov-verkada** [comment]
Addressed the comments.

File: docs/faq.md [hunk 3]
```diff
@@ -151,20 +151,20 @@ The default polling interval is 3 minutes (180 seconds) with a configurable jitt
 You can change the setting by updating the `timeout.reconciliation` value and the `timeout.reconciliation.jitter` in the [argocd-cm](https://github.com/argoproj/argo-cd/blob/2d6ce088acd4fb29271ffb6f6023dbb27594d59b/docs/operator-manual/argocd-cm.yaml#L279-L282) config map. If there are any Git changes, Argo CD will only update applications with the [auto-sync setting](user-guide/auto_sync.md) enabled. If you set it to `0` then Argo CD will stop polling Git repositories automatically and you can only use alternative methods such as [webhooks](operator-manual/webhook.md) and/or manual syncs for deploying applications.
 
 
-## Why Are My Resource Limits `Out Of Sync`?
+## Why is my ArgoCD application `Out Of Sync` and keep syncing when there are no actual changes in resource limits or other fields with unit values?
```
Comment (nitishfy) [inline docs/faq.md:154]:
```suggestion
## Why does my ArgoCD application keep syncing when there are no actual changes in resource limits or other fields with unit values?
```

how about this?

**Review by nitishfy — APPROVED** [review]
LGTM, just a minor nit.

File: docs/faq.md (hunk 3 above)
Comment (andrii-korotkov-verkada) [reply docs/faq.md:154]: I'd keep about out of sync, to also cover people who have auto sync disabled and still wondering about the issue.

Comment (blakepettersson) [reply docs/faq.md:154]: Or this one 

```suggestion
## Why is my ArgoCD application `Out Of Sync` when there are no actual changes to the resource limits (or other fields with unit values)?
```

Comment (andrii-korotkov-verkada) [reply docs/faq.md]: Done

**Review by blakepettersson — APPROVED** [review]

**Pull request merged**